packages/trpc/server/routers/viewer/oAuth/generateAuthCode.handler.ts

```diff
@@ -0,0 +1,80 @@
+import { randomBytes } from "crypto";
+
+import dayjs from "@calcom/dayjs";
+import { prisma } from "@calcom/prisma";
+import type { AccessScope } from "@calcom/prisma/enums";
+import type { TrpcSessionUser } from "@calcom/trpc/server/trpc";
+
+import { TRPCError } from "@trpc/server";
+
+import type { TGenerateAuthCodeInputSchema } from "./generateAuthCode.schema";
+
+type AddClientOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TGenerateAuthCodeInputSchema;
+};
+
+export const generateAuthCodeHandler = async ({ ctx, input }: AddClientOptions) => {
+  const { clientId, scopes, teamSlug } = input;
+  const client = await prisma.oAuthClient.findFirst({
+    where: {
+      clientId,
+    },
+    select: {
+      clientId: true,
+      redirectUri: true,
+      name: true,
+    },
+  });
+
+  if (!client) {
+    throw new TRPCError({ code: "UNAUTHORIZED", message: "Client ID not valid" });
+  }
+  const authorizationCode = generateAuthorizationCode();
+
+  const isTeam = teamSlug.startsWith("team/");
```

Comment:
Why do we need to do this? Just trying to understand when this is the case.

Reply:
Clean up the code a bit, I am now already making sure before that `teamSlug` only has the team slug and not also a username (https://github.com/calcom/cal.com/pull/11465/files#diff-77fc6507ac0ac4f357ba8d738052d1d61309bfc25c643599cfde2f344eefa565R165)